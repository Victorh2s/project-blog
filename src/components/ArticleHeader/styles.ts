import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`

  text-align: center;
  justify-content: center;
  margin-bottom: ${theme.spacings.xlarge};
  border-bottom: 0.1rem solid ${theme.colors.darkeyGray};

  ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }

  @media ${theme.media.lteMedium} {
      ${HeadingStyles}{
      font-size: ${theme.font.sizes.large};
    }
  }

`}
`;

export const Text = styled.p`
  text-align: center;
  justify-content: center;
`;

export const Img = styled.img`
  max-width: 100%;
  display: block;
`;
