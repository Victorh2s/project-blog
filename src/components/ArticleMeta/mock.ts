import { ArticleMetaProps } from '.';

export default {
  createdAt: '2022-06-10T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'victor-henrique',
    displayName: 'Victor H',
  },
  categories: [
    {
      id: 'asdslgyh65',
      displayName: 'Tech',
      slug: 'tech',
    },
    {
      id: 'asdslgyh66',
      displayName: 'JS',
      slug: 'javascript',
    },
  ],
} as ArticleMetaProps;
