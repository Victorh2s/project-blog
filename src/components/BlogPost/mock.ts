import { BlogPostProps } from '.';
import { data } from '../api/dados.json';

const { title, excerpt, content, cover, author, categories, createdAt } =
  data.posts[0];

export default {
  title,
  excerpt,
  content,
  cover,
  author,
  categories,
  createdAt,
} as BlogPostProps;
