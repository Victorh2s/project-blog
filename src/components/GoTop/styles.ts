import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    color: ${theme.colors.primary};
    border: 0.1rem solid;
    border-radius: 50rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
  `}
`;
