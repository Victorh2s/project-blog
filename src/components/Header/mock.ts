import { HeaderProps } from '.';

export default {
  BlogName: 'Usu√°rio teste',
  BlogDescription:
    'Et sed legere rationibus, vitae evertitur ad eam. Eu amet vero officiis vis. Ad nullam pericula neglegentur sit, vix ei regione legendos argumentum. Mei an quas convenire, nobis dicunt mei ut.',
  logo: 'assets/images/logo.png',
  showText: true,
} as HeaderProps;
