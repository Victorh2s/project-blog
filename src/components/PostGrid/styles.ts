import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  width: 100%;
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
  max-width: 120rem;
  max-height: 300rem;
  box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
  gap: ${theme.spacings.large};
  padding: ${theme.spacings.large};
  text-overflow: ellipsis;

  @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const NotFound = styled.p`
${({ theme }) => css`

    padding: ${theme.spacings.large};
    text-align: center;
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.secondary};

`}
`;
