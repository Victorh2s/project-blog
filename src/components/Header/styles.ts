import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';
import { Container as LogoLinkstyles } from '../LogoLink/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding-bottom: ${theme.spacings.huge};
    width: 100%;
    max-width: 96rem;
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;

    ${HeadingStyles} {
      margin: 0 0 calc(${theme.spacings.small} - 1rem);
    }

    ${LogoLinkstyles}{
      >img{
        border: 0.3rem solid;
        border-color:  rgb(74 222 128);
      }
    }

    @media ${theme.media.lteMedium}{
      display:flex;
      flex-flow: column wrap;

      & > ${HeadingStyles} {
        margin: 0 0 ${theme.spacings.medium} 0;
      }

    }
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;

    @media ${theme.media.lteMedium}{
      margin-left: 0;
    }
  `}
`;
