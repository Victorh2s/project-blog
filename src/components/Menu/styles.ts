import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

type MenuBehaviorProps = {
  Visible: boolean;
};

const ContainerChanger = (
  Visible: MenuBehaviorProps['Visible'],
  theme: DefaultTheme,
) => css`
  left: ${Visible ? 0 : '-35rem'};
  overflow-y: ${Visible ? 'auto' : 'hidden'};

  @media ${theme.media.lteMedium} {
    left: ${Visible ? '0' : '-35rem'};
  }
`;

export const Container = styled.div<MenuBehaviorProps>`
  ${({ theme, Visible }) => css`
    background: ${theme.colors.primary};
    padding: ${theme.spacings.large};
    display: flex;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    left: 0;
    transition: all 300ms ease-in-out;
    overflow-y: auto;
    ${ContainerChanger(Visible, theme)}
  `}
`;

export const Nav = styled.nav`
    margin: 0 auto;
    width: 100%;
`;

export const MenuLinks = styled.div`

   margin-bottom: 5rem;

`;

export const Logo = styled.div`

    ${HeadingStyles} {

      img{
        margin: 0 auto;
        margin-bottom: 3rem;
      }
    }

`;

const ButtonChanger = (
  Visible: MenuBehaviorProps['Visible'],
  theme: DefaultTheme,
) => css`
  left: ${Visible ? '26rem,' : '1rem'};
  color: ${Visible ? theme.colors.secondary : theme.colors.primary};

  @media ${theme.media.lteMedium} {
    left: ${Visible ? '26rem' : '-0.5rem'};
  }
`;

export const Button = styled.a<MenuBehaviorProps>`
  ${({ theme, Visible }) => css`
    position: fixed;
    top: ${theme.spacings.medium};
    left: 26rem;
    color: ${theme.colors.primary};
    z-index: 2;
    width: 3rem;
    height: 3rem;
    transition: all 300ms ease-in-out;
    ${ButtonChanger(Visible, theme)}
  `}
`;
