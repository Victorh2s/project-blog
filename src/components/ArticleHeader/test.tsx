import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { ArticleHeader } from '.';
import mock from './mock';
import { formatDate } from '../utils/format-date';

const props = mock;

describe('<ArticleHeader />', () => {
  it('should render Heading, Text and meta', () => {
    renderTheme(<ArticleHeader {...props} />);

    expect(
      screen.getByRole('heading', { name: props.title }),
    ).toBeInTheDocument();
    expect(screen.getByText(props.excerpt)).toBeInTheDocument();
    expect(screen.getByText(formatDate(props.createdAt))).toBeInTheDocument();
  });

  it('should render img', () => {
    renderTheme(<ArticleHeader {...props} />);

    expect(screen.getByRole('img')).toBeInTheDocument();
  });

  it('should match Snapshot', () => {
    const { container } = renderTheme(<ArticleHeader {...props} />);
    expect(container).toMatchSnapshot();
  });
});
