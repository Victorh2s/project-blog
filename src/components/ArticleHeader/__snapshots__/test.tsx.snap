// Vitest Snapshot v1

exports[`<ArticleHeader /> > should match Snapshot 1`] = `
.c2 {
  color: #0A1128;
  font-size: 4.0rem;
  text-transform: none;
}

.c0 {
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 4.0rem;
  border-bottom: 0.1rem solid;
}

.c0 .c1 {
  margin: 0;
  margin-bottom: 2.4rem;
}

.c3 {
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  max-width: 100%;
  display: block;
}

.c5 {
  font-size: 1.6rem;
  font-style: italic;
}

.c5 .categories span::after {
  content: ',';
}

.c5 .categories span:last-child::after {
  content: '';
}

.c5 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c5 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

@media (max-width:768px) {
  .c0 .c1 {
    font-size: 3.2rem;
  }
}

<div>
  <header
    class="c0"
  >
    <h1
      class="c1 c2"
    >
      Vis nobis veritus maluisset id, nonumy adversarium
    </h1>
    <p
      class="c3"
    >
      Fugit aliquip erroribus eu eam, an mel vidit doctus. Vis nobis veritus maluisset id, nonumy adversarium neglegentur est ut. Tation persius pri eu, modo minim fastidii mei no. Ad tota pertinax definiebas eam.
    </p>
    <img
      alt=""
      class="c4"
      src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614698965/Ubuntu_gel_by_Midge_Mantissa_Sinnaeve_7caea7ddb4.jpg"
    />
    <div
      class="c5"
    >
      <p>
        <span>
          Por 
        </span>
        <a
          href="/author/victor-henrique"
        >
          Victor H
        </a>
        <span
          class="separator"
        >
           | 
        </span>
        <time
          datetime="2022-06-10"
        >
          9 de jun. de 2022
        </time>
        <span
          class="separator"
        >
           | 
        </span>
        <span
          class="categories"
        >
          <span>
            <a
              href="/category/java-script"
            >
              JavaScript
            </a>
          </span>
          <span>
            <a
              href="/category/react"
            >
              React
            </a>
          </span>
        </span>
      </p>
    </div>
  </header>
</div>
`;
